### ISSUE TITLE
Adding Scons as a build system

### ISSUE THREAD

**Issue by adamjstewart** [LLNL/spack#986]
I'm thinking about adding Scons as a build system to Spack, similar to how Autotools and CMake are already integrated. This typically involves running `scons build [options]` and `scons install`, however, it can be done in parallel with `scons -j16 build [options]`. When running `spack create`, I would like it to detect whether or not the package should be built with Scons (as evidenced by the presence of a `SConstruct` file) and add these commands to the new package. It should obviously work with `spack install -j16` or `parallel = false`. And you shouldn't have to run `which(scons)` to run `scons('build', *options)`.

Can someone point me to all of the necessary files I need to edit?

**Comment by adamjstewart** [comment]
I added the following line to `lib/spack/spack/build_environment.py`:
```python
m.scons = MakeExecutable('scons', jobs)
```
That seems to handle `which(scons)` and all of the parallelization. But when I try building with it, I get the following error message:
```
Spack compiler must be run from Spack! Input 'SPACK_PREFIX' is missing.
```
Any idea why this isn't being set?

**Comment by adamjstewart** [comment]
Oh, I think I know what's going on here. From `scons help`:
```
* env_vars: [ string ]
    Environment variables to propagate through to SCons. Either the
    string "all" or a comma separated list of variable names, e.g.
    'LD_LIBRARY_PATH,HOME'.
    - default: 'LD_LIBRARY_PATH,PYTHONPATH'
```
So I just need to tell it to pass "all" environment variables to get the ones set by Spack.

Got `spack create` to generate a useful message for SCons packages.

**Comment by adamjstewart** [comment]
Closed by #994
